src/blockchain/blockchain_provider.ts

```diff
@@ -0,0 +1,37 @@
+import {
+  AmountType,
+  Box,
+  TokenId,
+  TransactionId,
+  UnsignedTransaction,
+} from "@fleet-sdk/common";
+import { Component } from "../component.ts";
+import { ErgomaticConfig } from "../config.ts";
+
+export interface BlockchainProvider {
+  submitTx(unsignedTx: UnsignedTransaction): Promise<TransactionId>;
+
+  getBoxByTokenId<T extends AmountType = string>(
+    tokenId: TokenId,
+  ): Promise<Box<T>>;
+}
+
+export class DefaultBlockchainProvider extends Component
```

Comment:
@arobsn I see nautilus uses a lot of gql, do you have any feelings on if we should go with that here or use REST api?

Reply:
Gonna go with REST api for now for the following reasons, can update later if you have other feelings:

- We will need REST for uexplorer, simpler to keep with one API type
- I dont have much gql experience so more familiar to me

Reply:
Oh, I'm sorry for the delay.

You are right. GraphQL is used in nautilus mostly because of cached responses and query flexibility, uexplorer will be a nice fit for ergomatic.